trace/report_buffer.go

```diff
@@ -85,7 +84,7 @@ func (buf *reportBuffer) addWitness(w *witnessWithInfo) {
 			// The witness exceeds our per-witness storage limit. Obfuscate it to
 			// reduce its size while retaining its typing information.
 			printer.Debugf("Obfuscating oversized witness (%d MB) captured on interface %s\n", len(witnessReport.WitnessProto)/1_000_000, w.netInterface)
-			data_masks.ZeroAllPrimitivesInMethod(w.witness.GetMethod())
+			buf.collector.redactor.ZeroAllPrimitives(w.witness.GetMethod())
```

Comment:
It's a little odd that we don't check whether it has already been obfuscated here, but this is OK.

Reply:
I think we record the state in `m.GetMeta().GetHttp().Obfuscation` on both paths?

Reply:
Added a check!